### PR TITLE
Workflow for Smoking test suite

### PR DESCRIPTION
**Pull request by liya2017**
test result: https://github.com/liya2017/ckb/runs/3131537694?check_suite_focus=true

### CODE REVIEW

File: devtools/somking_test/tip_block_growth_check.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    | tr -d '\n' \
+    | curl -H 'content-type: application/json' -d @- \
+    http://127.0.0.1:8114 `
+
+   TIP_BLOCK_NUMBER=`echo $tip_block_json | jq --raw-output '.result'`
+   TIP_BLOCK_NUMBER=$(printf %d $TIP_BLOCK_NUMBER)
+   echo $TIP_BLOCK_NUMBER
+}
+FIRST_TIP_BLOCK_NUMBER=`get_tip_block_number`
+echo "Fsirt tip block number is "$FIRST_TIP_BLOCK_NUMBER
```
Comment (yangby-cryptape) [inline devtools/somking_test/tip_block_growth_check.sh:23]:
```suggestion
echo "First tip block number is "$FIRST_TIP_BLOCK_NUMBER
```

File: .github/workflows/smoking_test.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  RUSTFLAGS: "-D warnings"
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+  TESTNET_SNAPSHOT_URL: ${{secrets.TESTNET_DATA_SNAPSHOT}}
+  CARGO_TARGET_DIR: ${{ github.workspace }}/../target
+
+jobs:
+   Start_and_sync:
+    runs-on: [self-hosted,Linux]
+    steps:
```
Comment (doitian) [inline .github/workflows/smoking_test.yaml:18]: I think we can use `workflow_dispatch` and `repository_dispatch` to trigger this workflow with the input to the CKB package URL. In `package.yml` we can trigger smoking test with the Linux package URL. So we can save the time to build CKB and just use the provided package to run the smoking test.

**Review by doitian — CHANGES_REQUESTED** [review]

**Review by keroro520 — DISMISSED** [review]
LGTM

**Comment by liya2017** [comment]
Add `repository_dispatch` based on Ian's comments

**Review by keroro520 — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
Blocking by reviewers @keroro520 @yangby-cryptape

**Review by keroro520 — APPROVED** [review]

**Comment by keroro520** [comment]
bors r=doitian,keroro520

**Pull request merged**